app/cni/pkg/cni/main_linux.go

```diff
@@ -66,239 +63,146 @@ type K8sArgs struct {
-		return nil, errors.Wrapf(err, "could not parse network configuration")
+		return nil, errors.Wrap(err, "failed to parse network configuration from stdin")
 	}
 
 	if conf.RawPrevResult != nil {
 		resultBytes, err := json.Marshal(conf.RawPrevResult)
 		if err != nil {
-			return nil, errors.Wrapf(err, "could not serialize prevResult")
+			return nil, errors.Wrap(err, "failed to serialize previous result")
 		}
+
 		res, err := version.NewResult(conf.CNIVersion, resultBytes)
 		if err != nil {
-			return nil, errors.Wrapf(err, "could not parse prevResult")
+			return nil, errors.Wrap(err, "failed to parse previous result")
 		}
+
 		conf.RawPrevResult = nil
-		conf.PrevResult, err = current.NewResultFromResult(res)
-		if err != nil {
-			return nil, errors.Wrapf(err, "could not convert result to current version")
+
+		if conf.PrevResult, err = current.NewResultFromResult(res); err != nil {
+			return nil, errors.Wrap(err, "failed to convert previous result to current version")
 		}
 	}
 
 	return &conf, nil
 }
 
-func hijackMainProcessStderr(logLevel string) (*os.File, error) {
+func hijackMainCNIProcessStderr(l logr.Logger) (*os.File, error) {
 	file, err := getCniProcessStderr()
 	if err != nil {
-		log.Error(err, "could not hijack main process file - continue logging to "+defaultLogLocation)
-		return nil, err
+		l.Error(err, "failed to hijack stderr of the CNI process, continuing to log to the default location: "+defaultLogLocation)
+		return nil, errors.Wrap(err, "unable to open CNI process stderr file")
 	}
-	log.V(0).Info("successfully hijacked stderr of cni process - logs will be available in 'kubectl logs'")
+
 	os.Stderr = file
-	if err := install.SetLogLevel(&log, logLevel, defaultLogName); err != nil {
-		return file, errors.Wrap(err, "wrong set the right log level")
-	}
 
-	return file, err
+	l.Info("successfully hijacked stderr of the CNI process; logs will be visible via 'kubectl logs'")
+
+	return file, nil
 }
 
 func getCniProcessStderr() (*os.File, error) {
-	pids, err := pidOf("/install-cni")
+	pid, err := pidOf(installCNIBinary)
 	if err != nil {
 		return nil, err
 	}
-	if len(pids) != 1 {
-		return nil, errors.New("more than one process '/install-cni' running on a node, this should not happen")
-	}
 
-	file, err := os.OpenFile(path.Join("/proc", strconv.Itoa(pids[0]), "fd", "2"), os.O_WRONLY, 0)
-	return file, err
+	return os.OpenFile(path.Join("/proc", pid, "fd", "2"), os.O_WRONLY, 0)
 }
 
 // cmdAdd is called for ADD requests
 func cmdAdd(args *skel.CmdArgs) error {
-	ctx := context.Background()
-	conf, err := parseConfig(args.StdinData)
-	if err != nil {
-		return errorLogged(log, err, "error parsing kuma-cni cmdAdd config")
-	}
-
-	mainProcessStderr, err := hijackMainProcessStderr(conf.LogLevel)
-	if mainProcessStderr != nil {
-		defer mainProcessStderr.Close()
-	}
+	conf, err := add(context.Background(), args)
 	if err != nil {
+		log.Info("[WARNING]: pod excluded", "reason", err)
```

Comment:
we don't usually log + return err. Is the err not logged in a func where cmdAdd is executed?

Reply:
For CNI we do it, so I didn't change this logic, just refactored it